array.h

```diff
@@ -1465,6 +1686,7 @@ void bubble_sort(Iterator begin, Iterator end, Compare&& comp) {
 // Sort the dims such that strides are increasing from dim 0, and contiguous
 // dimensions are fused.
 template <class Shape>
+NDARRAY_DECORATE_IF_CUDA
 shape_of_rank<Shape::rank()> dynamic_optimize_shape(const Shape& shape) {
```

Comment:
I wonder if this and dynamic_optimize_copy_shape below should be overloaded to do nothing on device. I'm not sure if they make sense to do in a cuda kernel. I imagine typically array operations using these in a kernel will be small, or should be done on the host prior to running the kernel.

Reply:
Resolved offline - keeping it since it *could* be used in a kernel, even if it's unlikely.